### PR TITLE
Print warning if both Brave and OTel are on the classpath

### LINKED ISSUE #36: Fail the context to start if both Brave and OTel is on the classpath
https://github.com/spring-cloud/spring-cloud-sleuth/issues/1855

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
fixes #36

### CODE REVIEW

File: spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ */
+class OtelMultipleTracersAnalyzer {
+
+	private static final Log log = LogFactory.getLog(OtelMultipleTracersAnalyzer.class);
+
+	@PostConstruct
+	void verify() {
+		if (isBraveOnClasspath()) {
+			log.warn(
+					"You have both Spring Cloud Sleuth OpenTelemetry (OTel) and Spring Cloud Sleuth Brave tracers on the classpath. We will pick OTel over Brave however please consider excluding <org.springframework.cloud:spring-cloud-sleuth-brave> dependency from your <org.springframework.cloud:spring-cloud-starter-sleuth> dependency.");
```
Comment (jonatan-ivanov) [inline spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java:42]: This is something like what I did when I played with this locally:

```
@ConditionalOnProperty(value = "spring.sleuth.enabled", matchIfMissing = true)
@ConditionalOnProperty(value = "spring.sleuth.conflictdetector", matchIfMissing = true) // TODO: how to name this?
@ConditionalOnClass(name =  "org.springframework.cloud.sleuth.autoconfig.otel.OtelAutoConfiguration")
```

So `@PostConstruct` does not need a condition.

Comment (jonatan-ivanov) [inline spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java:42]: I'm ok with a warn but would not it be better to throw an exception (and add the ability to turn it off)?

File: spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		}
+	}
+
+	boolean isBraveOnClasspath() {
+		return ClassUtils.isPresent("org.springframework.cloud.sleuth.autoconfig.brave.BraveAutoConfiguration", null);
+	}
+
+	static final class MultipleTracersFoundException extends RuntimeException {
```
Comment (jonatan-ivanov) [inline spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java:50]: Exception or `WARN`?

File: spring-cloud-sleuth-otel-autoconfigure/src/test/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzerTests.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		new OtelMultipleTracersAnalyzer() {
+			@Override
+			boolean isBraveOnClasspath() {
+				return true;
+			}
+		}.verify();
+
+		then(capturedOutput).contains("You have both Spring Cloud Sleuth OpenTelemetry");
+	}
+}
```
Comment (jonatan-ivanov) [inline spring-cloud-sleuth-otel-autoconfigure/src/test/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzerTests.java:54]: newline

**Review by jonatan-ivanov — CHANGES_REQUESTED** [review]

File: spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java (hunk 2 above)
Comment (marcingrzejszczak) [reply spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java:50]: I've forgotten to remove this

File: spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java (hunk 1 above)
Comment (marcingrzejszczak) [reply spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java:42]: The thing is that with the current implementation it's not a problem that you have 2 reporters at the classpath. But maybe indeed we could think of doing an opt-out exception throwing

Comment (marcingrzejszczak) [reply spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelMultipleTracersAnalyzer.java:42]: I did a fast POC of the feature. We can close this PR and please go ahead set it up properly with an opt-out version of throwing an exception when 2 tracers are on the classpath

**Pull request closed without merging**